frontend/src/features/admin-form/settings/components/EndPageSettingsSection/EndPageSettingsInput.tsx

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+import { FormControl, Stack } from '@chakra-ui/react'
+import validator from 'validator'
+
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import Textarea from '~components/Textarea'
+
+import { useMutateFormSettings } from '../../mutations'
+
+interface EndPageSettingsInputProps {
+  settings: {
+    title: string
+    paragraph?: string
+    buttonLink?: string
+    buttonText: string
+  }
+}
+
+export const EndPageSettingsInput = ({
+  settings,
+}: EndPageSettingsInputProps): JSX.Element => {
+  const { title, paragraph, buttonText, buttonLink } = settings
+  const { mutateFormEndPage } = useMutateFormSettings()
+
+  const defaultParagraph = useMemo(() => paragraph ?? '', [paragraph])
+  const defaultButtonLink = useMemo(() => buttonLink ?? '', [buttonLink])
+
+  const {
+    register,
+    formState: { errors },
+    resetField,
+    handleSubmit,
+  } = useForm<EndPageSettingsInputProps['settings']>({
+    mode: 'onChange',
+    defaultValues: {
+      title: title,
+      paragraph: defaultParagraph,
+      buttonText: buttonText,
+      buttonLink: defaultButtonLink,
+    },
+  })
+
+  const handleUpdateEndPage = useCallback(() => {
+    return handleSubmit((endPage) => {
+      if (
+        endPage.title === title &&
+        endPage.buttonText === buttonText &&
+        endPage.buttonLink === defaultButtonLink &&
+        endPage.paragraph === defaultParagraph
+      ) {
+        return
+      }
+
+      return mutateFormEndPage.mutate(endPage)
+    })()
```

Comment:
why is this invoked immediately?

Reply:
no need to invoke mutation immediately. can look at other examples in the code base as to how this is normally handled.

most of the time you return the mutation function and keep track of state using the mutation function's exported state (i.e. `mutateFormEndPage.isLoading`)